### PR TITLE
check rdkb wlan task state before handling steering socket remove event

### PR DESCRIPTION
**Pull request by kantera800**
Currently if any controller socket is disconnected an additional
STEERING_REMOVE_SOCKET event is sent to rdkb wlan task.
In case there are no actual steering listeners to remove this event
results in a failure and prints several error messages.
This happens each time any 3rd party API is triggered via BML and
fills the controller log file with unnecessary error messages.

Solution is to check if steering listener socket even exists before
trying to remove it.

### CODE REVIEW

**Review by arnout — APPROVED** [review]
Would be nice to have a generic way to do this, but OK.

**Review by tomereli — APPROVED** [review]

**Comment by kantera800** [comment]
@morantr added valuable (but offline) comment that it better to check if  socket exists.
made the changes and waiting for his approval.

File: controller/src/beerocks/master/tasks/rdkb/rdkb_wlan_task.cpp
Comment (morantr) [inline controller/src/beerocks/master/tasks/rdkb/rdkb_wlan_task.cpp:387]: Actually, I don't think this is necessary because it is already done in `remove_bml_rdkb_wlan_socket()`.
So I would recommend removing the function you added and remove also `set_bml_rdkb_wlan_events_update_enable()` since there is no point to set the 'events_update' flag if you remove the socket afterward on `remove_bml_rdkb_wlan_socket()`.
The only thing needs to be done here is to add return value (bool) to `remove_bml_rdkb_wlan_socket()` that indicates if the socket has been removed. If it has been removed, print the message to the log.

**Review by morantr — APPROVED** [review]
I think it can be better, but approving in order to no block this PR.

File: controller/src/beerocks/master/tasks/rdkb/rdkb_wlan_task.cpp
Comment (kantera800) [reply controller/src/beerocks/master/tasks/rdkb/rdkb_wlan_task.cpp:387]: This is a good point  , but then all steering APIs needs to be retested and this is not the purpose of this PR.
What i think needs to be done is to refactor RDKB task or to consider removing it completely (maybe move the implementation to rdkb wave api repo)

**Pull request merged**
